core/v1/meshfederation_types.go

```diff
@@ -80,14 +80,16 @@ type MeshFederationSecurity struct {
 	// +required
 	TrustDomain string `json:"trustDomain,omitempty"`
 
-	// Name of secret containing certificate chain to be used to validate
-	// the remote.  This is also used to validate certificates used by the
-	// remote services (both client and server certificates).
-	// XXX: maybe this is only used to initiate a connection, with the actual
-	// certs stored in the status field, as retrieved from the remote mesh, or
-	// maybe this identifies an endpoint used to retrieve a cert chain, a la jwks
+	// Name of ConfigMap containing certificate chain to be used to
+	// validate the remote.  This is also used to validate certificates used by
+	// the remote services (both client and server certificates).  The name of
+	// the entry should be of the form: <federation-name>-ca.crt, e.g. if the
+	// MeshFederation resource is named remote-mesh, the entry in the ConfigMap
+	// should be named remote-mesh-ca.crt.
+	// XXX: should we use the mesh federation name, or should we use the
+	// trustDomain as the key?
 	// +optional
-	CertificateChain string `json:"certificateChain,omitempty"`
+	CertificateChain corev1.TypedLocalObjectReference `json:"certificateChain,omitempty"`
```

Comment:
nit: should we rename the field to make it more obvious that it's referencing a ConfigMap?

Reply:
I left this in place in case we wanted to support secrets in the future.